temporian/implementation/numpy/data/event.py

```diff
@@ -25,81 +25,148 @@
     int: DType.INT64,
 }
 
+DTYPE_MAPPING = {
+    np.float64: DType.FLOAT64,
+    np.float32: DType.FLOAT32,
+    np.int64: DType.INT64,
+    np.int32: DType.INT32,
+    np.str_: DType.STRING,
+    np.string_: DType.STRING,
+    np.bool_: DType.BOOLEAN,
+}
+DTYPE_REVERSE_MAPPING = {
+    DType.FLOAT64: np.float64,
+    DType.FLOAT32: np.float32,
+    DType.INT64: np.int64,
+    DType.INT32: np.int32,
+    DType.STRING: np.str_,
+    DType.BOOLEAN: np.bool_,
+}
+
+
+@dataclass
+class IndexData:
+    features: List[np.ndarray]
+    timestamps: np.ndarray
+
+    def __init__(
+        self,
+        features: Union[np.ndarray, List[np.ndarray]],
+        timestamps: np.ndarray,
+    ) -> None:
+        if isinstance(features, np.ndarray):
+            features = [features]
+
+        self.features = features
+        self.timestamps = timestamps
+
+    def __len__(self) -> int:
+        return len(self.timestamps)
+
 
 class NumpyEvent:
+    # tolerance levels for comparing float features in __eq__
+    rtol: float = 1e-9
+    atol: float = 1e-9
+
     def __init__(
         self,
-        data: Dict[Tuple, List[NumpyFeature]],
-        sampling: NumpySampling,
+        data: Dict[Tuple, IndexData],
+        feature_names: Union[str, List[str]],
+        index_names: Union[str, List[str]],
+        is_unix_timestamp: bool = False,
     ) -> None:
         self._data = data
-        self._sampling = sampling
+        self._feature_names = (
+            [feature_names] if isinstance(feature_names, str) else feature_names
+        )
+        self._index_names = (
+            [index_names] if isinstance(index_names, str) else index_names
+        )
+        self._is_unix_timestamp = is_unix_timestamp
 
     @property
-    def data(self) -> Dict[Tuple, List[NumpyFeature]]:
+    def data(self) -> Dict[Tuple, IndexData]:
         return self._data
 
     @property
-    def sampling(self) -> NumpySampling:
-        return self._sampling
+    def feature_names(self) -> List[str]:
+        return self._feature_names
 
     @property
-    def _first_index_features(self) -> List[NumpyFeature]:
-        if self.first_index_value() is None:
-            return []
-        return self.data[self.first_index_value()]
+    def index_names(self) -> List[str]:
+        return self._index_names
+
+    @property
+    def is_unix_timestamp(self) -> bool:
+        return self._is_unix_timestamp
+
+    @property
+    def _first_index_features(self) -> List[np.ndarray]:
+        return list(self._data.values())[0].features
```

Comment:
I would go further. Since there is not guaranty that a feature exist or that there is an index (in both cases, this function will crash), I would remove the "_first_index_features" function and implement codes that currently call "_first_index_features" differently.